audb/core/api.py

```diff
@@ -113,7 +113,8 @@ def cached(
             flavor_id_paths = audeer.list_dir_names(version_path)
 
             # Skip old audb cache (e.g. 1 as flavor)
-            if audeer.list_file_names(version_path):  # pragma: no cover
+            files = audeer.list_file_names(version_path)
+            if len(files) > 1:  # pragma: no cover
```

Comment:
Not so nice, but we now have to allow one file as otherwise it will skip the root folder with the flavors as it now contains `db.csv`. Maybe we should find a nicer solution, but I propose to do that in a separate PR.

Reply:
Would it make sense to check if `db.csv` is included in `files` instead? At least it would maybe be easier to understand.
Otherwise I still your solution is still safe enough.

Reply:
> Would it make sense to check if `db.csv` is included in `files` instead? At least it would maybe be easier to understand.

That's what I tried first, but several tests failed. I didn't check why, though.

Reply:
Mh, strange and let we worry a little bit, but maybe just let's stick with your current solution then ;)

Reply:
Ah, I think I know why, I probably didn't expand `db.csv` to a full path.

Reply:
I tried with:

```python
            files = audeer.list_file_names(version_path)
            files = [os.path.basename(f) for f in files]
            if define.HEADER_FILE not in files:  # pragma: nocover
                continue
```
and it didn't work for me.

Reply:
Ah, my error I should not look for `define.HEADER_FILE` but for `define.DEPENDENCIES_FILE`

Reply:
Yes, the following works indeed for me:

```python
            files = audeer.list_file_names(version_path)
            files = [os.path.basename(f) for f in files]
            if define.DEPENDENCIES_FILE not in files:  # pragma: nocover
                continue
```
maybe we should add an argument to `list_dir_names()` and `list_file_names()` to ask for relative paths.

Reply:
Yes, I was missing such an argument a couple of times already.

Reply:
I created https://github.com/audeering/audeer/issues/37

Reply:
> Would it make sense to check if `db.csv` is included in `files` instead? At least it would maybe be easier to understand.

done

Reply:
Great your solution looks even simpler.